src/components/pagination/pagination.tsx

```diff
@@ -1,25 +1,30 @@
 import ReactPaginate from 'react-paginate';
-import { useDispatch, useSelector } from 'react-redux';
+import { useSelector } from 'react-redux';
 import RootState from '../../types/root-state';
-import { handleOffsetAction } from '../../state/action-creators';
+import { LOAD_MOVIES_AMOUNT } from '../../state/constants';
+import useGetParams from '../../hooks/use-get-params';
 import './style.css';
 
 export default function Pagination() {
-  const dispatch = useDispatch();
+  const { offset, setSearchParams } = useGetParams();
 
   const stateMoviesTotalAmount = useSelector(
     (state: RootState) => state.movieState.totalAmount
   );
 
-  const pageCount = Math.ceil(stateMoviesTotalAmount / 9);
+  const pageCount = Math.ceil(stateMoviesTotalAmount / LOAD_MOVIES_AMOUNT);
 
   const handlePageChange = (selected: number) =>
-    dispatch(handleOffsetAction(selected));
+    setSearchParams((params) => {
+      params.set('offset', selected.toString());
```

Comment:
it's better to avoid mutation and create a new object of params.

Reply:
Changed it, now it creates a new object from the url params and replaces the old ones from the state.